### PR TITLE
Hotpool v2 Imlementation

### LINKED ISSUE #1464: Hotpools
Tracker for all pieces needed for Hotpools

### PR DESCRIPTION
**Pull request by utopiabound**
Core logic to create Hotpool v2.

Fixes #1464

---
This change is [Reviewable](https://reviewable.io/reviews/whamcloud/integrated-manager-for-lustre/2241)

### CODE REVIEW

**Comment by utopiabound** [comment]
This is dependent on:
* #2178

File: chroma_core/models/filesystem.py [hunk 1]
```diff
@@ -109,6 +109,20 @@ def get_servers(self):
...
+
+        groups = {}
+
+        for t in targets:
+            hosts = [x.host for x in t.managedtargetmount_set.all()]
+            name = ",".join(sorted([x.fqdn for x in hosts]))
+            if name not in groups:
+                groups[name] = hosts
+
+        return groups.values()
```
Comment (jgrund) [inline chroma_core/models/filesystem.py:124]: This fn isn't accurate as it doesn't account for bans.

File: chroma_api/hotpool.py [hunk 2]
```diff
@@ -0,0 +1,96 @@
+# Copyright (c) 2020 DDN. All rights reserved.
+# Use of this source code is governed by a MIT-style
+# license that can be found in the LICENSE file.
+
+from collections import defaultdict
```
Comment (jgrund) [inline chroma_api/hotpool.py:5]: The API should be part of graphql instead.

File: chroma_core/services/job_scheduler/job_scheduler.py [hunk 3]
```diff
@@ -1992,3 +1999,158 @@ def remove_task(self, task_id):
 
         self.progress.advance()
         return task.id, command_id
+
+    def create_hotpool(self, hotpool_data):
```
Comment (jgrund) [inline chroma_core/services/job_scheduler/job_scheduler.py:2003]: This should be a graphql endpoint.

**Review by jgrund — CHANGES_REQUESTED** [review]
There are a few things in here that should leverage graphql instead.

**Review by jgrund — CHANGES_REQUESTED** [review]
Python hotpool api still needs removal

File: chroma_core/services/job_scheduler/job_scheduler.py (hunk 3 above)
Comment (jgrund) [reply chroma_core/services/job_scheduler/job_scheduler.py:1991]: Unresolving as current patch still contains change.

File: iml-api/src/graphql/mod.rs [hunk 4]
```diff
@@ -250,6 +272,49 @@ impl QueryRoot {
         Ok(xs)
     }
 
+    #[graphql(arguments(
+        limit(description = "paging limit, defaults to 20"),
+        offset(description = "Offset into items, defaults to 0"),
+        dir(description = "Sort direction, defaults to asc"),
+    ))]
+    /// Fetch the list of known targets
```
Comment (jgrund) [inline iml-api/src/graphql/mod.rs:280]: Update docs to describe params

File: iml-wire-types/src/lib.rs [hunk 5]
```diff
@@ -1997,6 +1997,37 @@ pub struct FidItem {
...
+    pub version: Option<i32>,
+
+    /// minimum age before mirror to cold pool
+    pub minage: i32,
+    pub freehi: Option<i32>,
+    pub freelo: Option<i32>,
+    /// Hot OstPool
+    pub hot_id: i32,
+    /// Cold OstPool
+    pub cold_id: i32,
```
Comment (jgrund) [inline iml-wire-types/src/lib.rs:2022]: Update to `cold_pool_name`

File: iml-wire-types/src/lib.rs [hunk 6]
```diff
@@ -1997,6 +1997,37 @@ pub struct FidItem {
...
+    pub ha_label: String,
+    /// Version of Hotpool Config
+    pub version: Option<i32>,
+
+    /// minimum age before mirror to cold pool
+    pub minage: i32,
+    pub freehi: Option<i32>,
+    pub freelo: Option<i32>,
+    /// Hot OstPool
+    pub hot_id: i32,
```
Comment (jgrund) [inline iml-wire-types/src/lib.rs:2020]: Update to `hot_pool_name`

**Pull request closed without merging**